gamemode/modules/commands/sv_commands.lua

```diff
@@ -0,0 +1,314 @@
+            table.insert( selectedPlayers, ply )
+            playerList = playerList .. "\n" .. plyNick
+        end
+    end
+
+    if #selectedPlayers > 1 then
+        return nil, "\"" .. arg .. "\" matches multiple players:" .. playerList
+    elseif #selectedPlayers == 1 then
+        return selectedPlayers[1]
+    end
+
+    return nil, "Invalid player: " .. arg
+end )
+
+function commands.checkArguments( argNames, argTypes, args )
+    if #args < #argTypes then
+        local commandHelp = ""
+
+        for i, argName in ipairs( argNames ) do
+            local argType = argTypes[i]
+
+            commandHelp = commandHelp .. argName .. ":" .. argType .. " "
+        end
+
+        return nil, "Missing argument. Usage " .. commandHelp
+    end
+
+    local newArgs = {}
+    local allPlayersReferenced = false
+
+    for i, argType in ipairs( argTypes ) do
+        local arg = args[i]
+
+        local value, errorMsg = commands.types[argType]( arg )
+
+        if errorMsg then
+            return nil, errorMsg
+        end
+
+        if argType == "player" and value == "@" then
+            if allPlayersReferenced then
+                return nil, "Cannot use @ more than once on a command"
+            end
+
+            allPlayersReferenced = true
+        end
+
+        newArgs[i] = value
+    end
+
+    return newArgs
+end
+
+local function processArguments( argsStr )
+    argsStr = string.Trim( argsStr )
+
+    local args = {}
+    local str = ""
+
+    local insideQuotes = false
+
+    local i = 1
+
+    while i <= string.len( argsStr ) do
+        local char = string.sub( argsStr, i, i )
+
+        if char == "\"" then
+            insideQuotes = not insideQuotes
+        elseif char == "\\" then
+            i = i + 1
+
+            char = string.sub( argsStr, i, i )
+            str = str .. char
+        elseif char == " " and not insideQuotes then
+            table.insert( args, str )
+            str = ""
+        else
+            str = str .. char
+        end
+
+        i = i + 1
+    end
+
+    if str ~= "" then
+        table.insert( args, str )
+    end
+
+    return args
+end
+
+local function processCommand( ply, command, args )
+    local commandInfo = commands.commands[command]
+
+    if commandInfo then
+        if not RVR.isUserGroup( ply, commandInfo.userGroup ) then
+            return "You need to be " .. RVR.getGroupName( commandInfo.userGroup ) .. " to use this command"
+        end
+
+        local newArgs, errorMsg = commands.checkArguments( commandInfo.argNames, commandInfo.argTypes, args )
+
+        if errorMsg then
+            return errorMsg, true
+        end
+
+        local hasAllPlayerReference = false
+        local allPlayerReferenceLocation = nil
+
+        for i, arg in ipairs( newArgs ) do
+            if arg == "^" and commandInfo.argTypes[i] == "player" then
+                newArgs[i] = ply
+            end
+
+            if arg == "@" and commandInfo.argTypes[i] == "player" then
```

Comment:
`@` in ULX doesn't get all players, it gets the calling player's aim entity.
`*` is what get's all players, and if we want to support that, we should have an argument type called `players` which passes in a list of players, rather than calling the command for every player. Would mean `*` could be used in multiple places in the command, as you just replace it with `player.GetAll()`. Would also mean we can control the print for commands that do things to multiple players more neatly.